### PR TITLE
Update godoc

### PR DESCRIPTION
**Pull request by colin-axner**

### CODE REVIEW

**Comment by AdityaSripal** [comment]
Looks like there's a build error on this branch

File: app/genesis.go [hunk 1]
```diff
@@ -4,15 +4,19 @@ import (
 	"github.com/tendermint/tendermint/crypto"
 )
 
+// GenesisState specifies the validator of the chain
 type GenesisState struct {
 	Validator GenesisValidator `json:"validator"`
 }
 
+// GenesisValidator holds the consensus public key and fee address of
```
Comment (AdityaSripal) [inline app/genesis.go]: Should note that ConsPubKey is tendermint `Ed25119` keys (check this first i forget exact sig algorithm we use).
While the Fee Address is an Ethereum address

File: cmd/plasmacli/store/keystore.go [hunk 2]
```diff
@@ -40,8 +40,8 @@ func InitKeystore(homeDir string) {
 	}
 }
 
-// Return iterator over accounts
-// returns db so db.close can be called
+// AccountIterator returns an iterator for accounts. It returns db so
+// db.close can be called.
```
Comment (AdityaSripal) [inline cmd/plasmacli/store/keystore.go]: This should be a contract.
```suggestion
// CONTRACT: Caller is responsible for closing db after use.
```

File: cmd/plasmacli/subcmd/eth/challenge.go [hunk 3]
```diff
@@ -14,6 +14,7 @@ import (
 	"strconv"
 )
 
+// ChallengeCmd returns the eth challenge command
```
Comment (AdityaSripal) [inline cmd/plasmacli/subcmd/eth/challenge.go]: For all CMD documentation. I think more detail is needed.
Suggested pattern.
```suggestion
// ChallengeCmd returns the eth challenge command. Use to challenge invalid exits.
```

^Some basic description of what the command does should be added to all cmd documentation imo. Can be a single sentence or two

File: handlers/errors.go [hunk 4]
```diff
@@ -15,26 +16,33 @@ const (
...
 func ErrSignatureVerificationFailure(msg string, args ...interface{}) sdk.Error {
 	return sdk.NewError(DefaultCodespace, CodeSignatureVerificationFailure, msg, args)
 }
 
+// ErrInvalidTransaction error for an invalid transaction
 func ErrInvalidTransaction(msg string, args ...interface{}) sdk.Error {
 	return sdk.NewError(DefaultCodespace, CodeInvalidTransaction, msg, args)
 }
 
+// ErrInvalidSignture error for an incorrect signture
```
Comment (AdityaSripal) [inline handlers/errors.go]:
```suggestion
// ErrInvalidSignature error for an incorrect signature
```

File: msgs/txDecoder.go [hunk 5]
```diff
@@ -6,6 +6,8 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
+// TxDecoder attempts to RLP decode the transaction bytes into a SpendMsg or
```
Comment (AdityaSripal) [inline msgs/txDecoder.go]:
```suggestion
// TxDecoder attempts to RLP decode the transaction bytes into a SpendMsg first then to a
```

File: app/genesis.go [hunk 6]
```diff
@@ -4,15 +4,19 @@ import (
...
+// the validator. ConsPubKey is tendermint Ed25119 public key.
```
Comment (AdityaSripal) [inline app/genesis.go]:
```suggestion
// the validator. ConsPubKey is tendermint Ed25119 public key for signing consensus messages, FeeAddress is Ethereum address for collecting fees.
```

File: cmd/plasmacli/store/keystore.go [hunk 7]
```diff
@@ -40,8 +40,8 @@ func InitKeystore(homeDir string) {
 	}
 }
 
-// AccountIterator returns an iterator for accounts. It returns db so
-// db.close can be called.
+// AccountIterator returns an iterator for accounts. CONTRACT: Caller is responsible
```
Comment (AdityaSripal) [inline cmd/plasmacli/store/keystore.go]: Put CONTRACT on new line

File: cmd/plasmacli/store/keystore.go [hunk 8]
```diff
@@ -53,8 +53,8 @@ func AccountIterator() (iterator.Iterator, *leveldb.DB) {
 	return db.NewIterator(nil, nil), db
 }
 
-// Add a new account to the keystore
-// Add account name and address to leveldb
+// AddAccount adds a new account to the keystore. It adds the account name
+// and address to leveldb.
```
Comment (hamdiallam) [inline cmd/plasmacli/store/keystore.go]: s/leveldb/db.

The Account wrapper should be agnostic to the backend db

File: cmd/plasmacli/store/keystore.go [hunk 9]
```diff
@@ -105,8 +105,7 @@ func GetAccount(name string) (ethcmn.Address, error) {
 	return ethcmn.BytesToAddress(addr), nil
 }
 
-// Remove an account from the local keystore
-// and the leveldb
+// DeleteAccount removes an account from the local keystore and the leveldb.
```
Comment (hamdiallam) [inline cmd/plasmacli/store/keystore.go]: ffti: end after "removes an account". or "removes an account from db"

File: cmd/plasmacli/store/keystore.go [hunk 10]
```diff
@@ -139,7 +138,8 @@ func DeleteAccount(name string) error {
 	return nil
 }
 
-// Update either the name of an account or the passphrase for an account
+// UpdateAccount updates either the name of an account or the passphrase for
+// an account.
```
Comment (hamdiallam) [inline cmd/plasmacli/store/keystore.go]: I don't see how this updates the passphrase.

File: cmd/plasmacli/store/keystore.go [hunk 11]
```diff
@@ -187,7 +187,7 @@ func UpdateAccount(name string, updatedName string) (msg string, err error) {
 	return msg, nil
 }
 
-// Import a private key with an account name
+// ImportECDSA imports a private key with an account name.
```
Comment (hamdiallam) [inline cmd/plasmacli/store/keystore.go]: private key associated with an account.

File: cmd/plasmacli/store/sigs.go [hunk 12]
```diff
@@ -10,11 +10,10 @@ const (
 	signatureDir = "data/signatures.ldb"
 )
 
-// Saves confirmation signatures
-// If prepend is set to true, sig will be prepended to the currently stored signatures
-// Otherwise it will be appended
-// This ordering should be determined by input order in the transaction
-// If the length of the currently stored signatures is 130 an error is returned
+// SaveSig saves the confirmation signatures.If prepend is set to true, sig
```
Comment (hamdiallam) [inline cmd/plasmacli/store/sigs.go]: space after the period

File: cmd/plasmad/config/config.go [hunk 13]
```diff
@@ -81,14 +83,16 @@ func TestPlasmaConfig() PlasmaConfig {
 	}
 }
 
-// parses the plasma.toml file and unmarshals it into a Config struct
+// ParsePlasmaConfigFromViper parses the plasma.toml file and unmarshals it
+// into a Config struct.
```
Comment (hamdiallam) [inline cmd/plasmad/config/config.go]: s/Config/PlasmaConfig

File: store/outputs.go [hunk 14]
```diff
@@ -110,128 +110,128 @@ func (store DataStore) GetTx(ctx sdk.Context, hash []byte) (Transaction, bool) {
...
 }
 
 // StoreTx adds the transaction.
-func (store DataStore) StoreTx(ctx sdk.Context, tx Transaction) {
-	store.setTx(ctx, tx)
+func (ds DataStore) StoreTx(ctx sdk.Context, tx Transaction) {
+	ds.setTx(ctx, tx)
 }
 
 // StoreOutputs adds new Output UTXO's to respective owner wallets.
```
Comment (hamdiallam) [inline store/outputs.go]: *the respective

File: store/outputs.go [hunk 15]
```diff
@@ -304,13 +304,13 @@ func (store DataStore) SpendOutput(ctx sdk.Context, pos plasma.Position, spender
 // GetUnspentForWallet returns the unspent outputs that belong to the given
```
Comment (hamdiallam) [inline store/outputs.go]: so the user*

File: store/outputs.go [hunk 16]
```diff
@@ -304,13 +304,13 @@ func (store DataStore) SpendOutput(ctx sdk.Context, pos plasma.Position, spender
...
 // wallet. Returns the struct TxOutput so user has access to the
```
Comment (hamdiallam) [inline store/outputs.go]: so the user*

File: cmd/plasmacli/store/keystore.go (hunk 10 above)
Comment (colin-axner) [reply cmd/plasmacli/store/keystore.go]: the else statement ` if err = ks.Update(acc, password, updatedPassword); err != nil {`

**Pull request merged**